### PR TITLE
homeScreen

### PR DESCRIPTION
**Pull request by AbdelrahmanMohamed-m**

### CODE REVIEW

File: core/data/src/main/java/org/the_chance/honeymart/data/repository/HoneyMartRepositoryImp.kt [hunk 1]
```diff
@@ -107,22 +109,19 @@ class HoneyMartRepositoryImp @Inject constructor(
     override suspend fun deleteAllCart(): String =
         wrap { honeyMartService.deleteAllFromCart() }.value ?: throw NotFoundException()
 
-    override suspend fun getUSerCoupons(): List<CouponEntity> =
-        wrap { honeyMartService.getCouponOfUser() }.value?.map { it.toCouponEntity() }
+    override suspend fun getAllUserCoupons(): List<CouponEntity> =
```
Comment (AmeerAmjed) [inline core/data/src/main/java/org/the_chance/honeymart/data/repository/HoneyMartRepositoryImp.kt:112]: Don't use compact function with multi line

File: core/data/src/main/java/org/the_chance/honeymart/data/source/remote/network/HoneyMartService.kt [hunk 2]
```diff
@@ -158,10 +157,12 @@ interface HoneyMartService {
 
 
     // region Coupon
-    suspend fun getCouponOfUser(): BaseResponse<List<CouponDto>>
+    suspend fun getAllUserCoupons(): BaseResponse<List<CouponDto>>
```
Comment (AmeerAmjed) [inline core/data/src/main/java/org/the_chance/honeymart/data/source/remote/network/HoneyMartService.kt:160]: I don't understand what is functionally this function need improve name

File: user/src/main/java/org/the_chance/honeymart/ui/feature/bottom_navigation/BottomBarItems.kt [hunk 3]
```diff
@@ -11,7 +11,7 @@ sealed class BottomBarItems(
 ) {
     object Home : BottomBarItems(
         route = Graph.HOME,
-        label = "Market",
+        label = "Home",
```
Comment (AmeerAmjed) [inline user/src/main/java/org/the_chance/honeymart/ui/feature/bottom_navigation/BottomBarItems.kt:14]: Why hardcode

File: user/src/main/java/org/the_chance/honeymart/ui/feature/home/HomeScreen.kt [hunk 4]
```diff
@@ -29,35 +30,38 @@ fun HomeScreen(
...
-        listener = viewModel
-    )
-//    LaunchedEffect(Unit) {
-//        while (true) {
-//            delay(3000)
-//            pagerState.animateScrollToPage(page = (pagerState.currentPage + 1) % 3)
-//        }
-//    }
-
     LaunchedEffect(key1 = true) {
```
Comment (AmeerAmjed) [inline user/src/main/java/org/the_chance/honeymart/ui/feature/home/HomeScreen.kt:33]: Should be in the end screen content

File: user/src/main/java/org/the_chance/honeymart/ui/feature/home/HomeUiState.kt [hunk 5]
```diff
@@ -1,62 +1,79 @@
...
-import org.the_chance.honeymart.ui.feature.orders.OrderStates
 import org.the_chance.honeymart.ui.feature.orders.OrderUiState
 import org.the_chance.honeymart.ui.feature.product.ProductUiState
+import org.the_chance.honeymart.ui.feature.product.toProductUiState
 
 data class HomeUiState(
-    val searchClick: Boolean = false,
     val isLoading: Boolean = true,
-    val isError: Boolean = false,
+    val isConnectionError: Boolean = false,
```
Comment (AmeerAmjed) [inline user/src/main/java/org/the_chance/honeymart/ui/feature/home/HomeUiState.kt:15]: Why error to specific, why not general.

File: user/src/main/java/org/the_chance/honeymart/ui/feature/home/composables/coupon/CouponDetailsShape.kt [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+            couponDetailsPath(
+                size = size,
+                cornerRadius = CornerSize(cornerRadius).toPx(size, density),
+                notchRadius = CornerSize(notchRadius).toPx(size, density)
+            )
+        )
+    }
+}
+
+fun couponDetailsPath(size: Size, cornerRadius: Float, notchRadius: Float): Path {
```
Comment (AmeerAmjed) [inline user/src/main/java/org/the_chance/honeymart/ui/feature/home/composables/coupon/CouponDetailsShape.kt:33]: Why function not private

File: user/src/main/java/org/the_chance/honeymart/ui/feature/home/composables/coupon/CouponsItem.kt [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+        CouponImage(
+            modifier = Modifier.fillMaxHeight(),
+            productImageUrl = coupon.product.productImages.takeIf { it.isNotEmpty() }?.get(0) ?: "",
+            couponCode = coupon.couponId.toString(),
+        )
+    }
+}
+
+@Composable
+fun CouponDetails(
```
Comment (AmeerAmjed) [inline user/src/main/java/org/the_chance/honeymart/ui/feature/home/composables/coupon/CouponsItem.kt:57]: Why multi composable in one file with multi type

**Review by AmeerAmjed — CHANGES_REQUESTED** [review]

**Review by Eng-Light — APPROVED** [review]

File: user/src/main/java/org/the_chance/honeymart/ui/feature/home/composables/HomeCategoriesItem.kt [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+                    it.drawOutline(
+                        outline = Outline.Generic(path),
+                        paint = Paint().apply {
+                            color = colors.secondaryContainer
+                            pathEffect = PathEffect.cornerPathEffect(16.dp.toPx())
+                        }
+                    )
+                }
+            }
+            .width(130.dp)
```
Comment (TarekIdrees) [inline user/src/main/java/org/the_chance/honeymart/ui/feature/home/composables/HomeCategoriesItem.kt:64]: replace it with the material.dimens

File: user/src/main/java/org/the_chance/honeymart/ui/feature/home/composables/LastPurchases.kt [hunk 9]
```diff
@@ -31,8 +31,7 @@ fun LastPurchasesItems(
     image: String = "",
     label: String = "blah blah blah",
     onClick: () -> Unit = {},
-
-    ) {
+) {
     Box(
         modifier = modifier
             .size(width = 149.dp, height = 152.dp)
```
Comment (TarekIdrees) [inline user/src/main/java/org/the_chance/honeymart/ui/feature/home/composables/LastPurchases.kt:37]: replace it with the material.dimens

File: user/src/main/java/org/the_chance/honeymart/ui/feature/home/composables/ProductItem.kt [hunk 10]
```diff
@@ -25,17 +25,16 @@ import org.the_chance.honymart.ui.theme.dimens
...
+    productPrice: String,
     imageUrl: String,
-    onClick: () -> Unit ,
-
-    ) {
+    onClick: () -> Unit,
+) {
     Box(
         modifier = modifier
             .size(width = 160.dp, height = 136.dp)
```
Comment (TarekIdrees) [inline user/src/main/java/org/the_chance/honeymart/ui/feature/home/composables/ProductItem.kt:40]: replace it with the material.dimens

File: user/src/main/java/org/the_chance/honeymart/ui/feature/home/composables/ProductItem.kt [hunk 11]
```diff
@@ -25,17 +25,16 @@ import org.the_chance.honymart.ui.theme.dimens
 import org.the_chance.honymart.ui.theme.white
 
 @Composable
-fun NewProductsItems(
+fun ProductItem(
     modifier: Modifier = Modifier,
```
Comment (TarekIdrees) [inline user/src/main/java/org/the_chance/honeymart/ui/feature/home/composables/ProductItem.kt:29]: the modifier parameter should be the first parameter with the default value, to be like this : 
isFavoriteIconClicked: Boolean,
onClickFavorite: () -> Unit,
productName: String,
productPrice: String,
modifier: Modifier = Modifier,
enable: Boolean = true,

**Review by TarekIdrees — CHANGES_REQUESTED** [review]

File: user/src/main/java/org/the_chance/honeymart/ui/feature/home/composables/HomeCategoriesItem.kt [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+            .height(120.dp),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.Center,
+    ) {
+        /*ImageNetwork(
```
[the remark this answers is not in the document]
Comment (AbdelrahmanMohamed-m) [reply user/src/main/java/org/the_chance/honeymart/ui/feature/home/composables/HomeCategoriesItem.kt:69]: Left to when categories images is added

**Review by AmeerAmjed — APPROVED** [review]
Good job 👏😍👏.

**Review by Eng-Light — APPROVED** [review]

**Pull request merged**